sources/web/datalab/polymer/modules/Utils.ts

```diff
@@ -41,32 +37,40 @@ class Utils {
 
   /**
    * Opens a dialog with the specified options. It uses the Datalab custom element
-   * <input-dialog>, attaches a new instance to the current document, opens it
-   * and returns a promise that resolves when the dialog is closed.
-   * @param dialogOptions options for configuring the dialog
+   * according to the specified dialog type, attaches a new instance to the current
+   * document, opens it, and returns a promise that resolves when the dialog is closed.
+   * @param type specifies which type of dialog to use
+   * @param dialogOptions specifies different options for opening the dialog
    */
-  static getUserInputAsync(dialogOptions: DialogOptions) {
-    const createModal = <InputDialogElement>document.createElement('input-dialog');
-    document.body.appendChild(createModal);
+  static showDialog(type: DialogType, dialogOptions: DialogOptions) {
+    let dialogElement = '';
+    if (type === DialogType.input) {
```

Comment:
I'm not sure I see how this is better, I think a simple enum is all we need here.
I'd love to see your take on this, let's discuss this face to face.